File: Ridy-taxi-app/ios/common/Networking/WebSocket/GetPaymentLink.swift
//
//  GetPaymentLink.swift
//  rider
//
//  Created by Manly Man on 9/10/20.
//  Copyright © 2020 minimal. All rights reserved.
//

import Foundation

class GetPaymentLink: SocketRequest {
    typealias ResponseType = GetPaymentLinkResult
    var params: [Any]?
    
    init(gatewayId: Int, amount: Double, currency: String) {
        let dto = GetPaymentLinkDTO(gatewayId: gatewayId, amount: amount, currency: currency, serverUrl: Config.Backend)
        let dic = try! dto.asDictionary()
        print(dic)
        self.params = [dic]
    }
}

struct GetPaymentLinkDTO: Codable {
    var gatewayId: Int
    var amount: Double
    var currency: String
    var serverUrl: String
}

struct GetPaymentLinkResult: Codable {
    var url: String
}

File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/networking/socket/GetPaymentLink.kt
package com.innomalist.taxi.common.networking.socket

import com.innomalist.taxi.common.networking.socket.interfaces.SocketRequest
import com.innomalist.taxi.common.utils.Adapters
import org.json.JSONObject

class GetPaymentLink(gatewayId: Int, currency: String, amount: Double, serverUrl: String): SocketRequest() {
    init {
        val dto = GetPaymentLinkDTO(gatewayId, amount, currency, serverUrl)
        val obj = JSONObject(Adapters.moshi.adapter(GetPaymentLinkDTO::class.java).toJson(dto))
        this.params = arrayOf(obj)
    }
}

data class GetPaymentLinkDTO (
        val gatewayId: Int,
        val amount: Double,
        val currency: String,
        val serverUrl: String
)

data class GetPaymentLinkResult (
        val url: String
)